src/map/script.c

```diff
@@ -24559,6 +24559,28 @@ static BUILDIN(getcalendartime)
 	return true;
 }
 
+static BUILDIN(setfavoriteitem)
+{
+	struct map_session_data *sd = script_rid2sd(st);
+	int idx = script_getnum(st, 2);
+	int value = script_getnum(st, 3);
+
+	if (sd == NULL) {
+		ShowError("buildin_setfavoriteitem: no player attached.\n");
+		return false;
+	}
+	else if (idx < 0 || idx >= sd->status.inventorySize || sd->status.inventorySize <= 0) {
```

Comment:
``` } else if (x) { ```

Reply:
Also there's no need to check the inventory size, the index check is enough to validate it

Reply:
done